code/src/App.js

```diff
@@ -1,9 +1,83 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react';
+
+import { HAPPY_THOUGHTS_URL, HEARTS_URL } from './reusable/urls';
+
+import MessageForm from 'components/MessageForm';
+import MessageList from 'components/MessageList';
 
 export const App = () => {
+  const [messageList, setMessageList] = useState([]);
+  const [messageNew, setMessageNew] = useState(['']);
+
+  useEffect(() => {
+    fetchMessageList();
+  }, []);
+
+  const fetchMessageList = () => {
+    fetch(HAPPY_THOUGHTS_URL)
+      .then(res => res.json())
+      .then(inputs => setMessageList(inputs))
+      .catch(err => console.error(err));
+  }
+
+  const handleMessageNewChange = (e) => {
+    setMessageNew(e.target.value);
+  }
+
+  const handleFormSubmit = (e) => {
+    e.preventDefault();
+
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ message: messageNew })
+    }
+
+    fetch(HAPPY_THOUGHTS_URL, options)
+      .then(res => res.json())
+      .then(recievedMessage => setMessageList([...messageList, recievedMessage]))
```

Comment:
What you also can do is to in the 2nd .then()  send the input directly to the `fetchMessageList()`. It makes the code cleaner. Eg: `.then(() => fetchMessageList())`. Then you refetch the data from the server instead of updating the local messageList state. So everytime you do a POST Request you also do a GET request for the new data. Maks explained it good in the last lecture during wednesday! Have a look at the lecture and try it out maybe when you come back to your project 🦄 👼